analyzers/src/SonarAnalyzer.CSharp/Rules/CloudNative/AzureFunctionsLogFailures.cs

```diff
@@ -0,0 +1,189 @@
+            public ImmutableArray<ExpressionSyntax> InvalidLoggerInvocations => invalidIvocationsBuilder.ToImmutableArray();
+
+            public override void Visit(SyntaxNode node)
+            {
+                CancellationToken.ThrowIfCancellationRequested();
+                if (HasValidLoggerCall)
+                {
+                    return;
+                }
+                base.Visit(node);
+            }
+
+            public override void VisitInvocationExpression(InvocationExpressionSyntax node)
+            {
+                if (Model.GetSymbolInfo(node, CancellationToken).Symbol is IMethodSymbol { ReceiverType: { } receiver }
+                    && receiver.DerivesOrImplements(ILogger))
+                {
+                    if (IsValidLogCall(node))
+                    {
+                        HasValidLoggerCall = true;
+                    }
+                    else
+                    {
+                        invalidIvocationsBuilder.Add(node);
+                    }
+                }
+                base.VisitInvocationExpression(node);
+            }
+
+            private bool IsValidLogCall(InvocationExpressionSyntax invocation)
+            {
+                if (Model.GetSymbolInfo(invocation, CancellationToken).Symbol is IMethodSymbol symbol)
+                {
+                    if (symbol.IsExtensionMethod
+                        && LoggerExtensions.Value is { } loggerExtensions
+                        && loggerExtensions.Equals(symbol.ContainingType)
+                        && (symbol.Name is "LogWarning" or "LogError" or "LogCritical"
+                           || (symbol.Name is "Log" && IsPassingValidLogLevel(invocation, symbol))))
+                    {
+                        return true;
+                    }
+                    else
+                    {
+                        if (ILogger_Log.Value is { } loggerLog
+                            && (symbol.OriginalDefinition.Equals(loggerLog)
+                                || symbol.ContainingType.FindImplementationForInterfaceMember(loggerLog)?.Equals(symbol.OriginalDefinition) == true))
+                        {
+                            return IsPassingValidLogLevel(invocation, symbol);
+                        }
+                    }
+                }
+                return false;
+            }
+
+            private bool IsPassingValidLogLevel(InvocationExpressionSyntax invocation, IMethodSymbol symbol) =>
+                symbol.Parameters.FirstOrDefault(x => x.Name == "logLevel") is { } logLevelParameter
+                    && Language.MethodParameterLookup(invocation, symbol).TryGetNonParamsSyntax(logLevelParameter, out var argumentSyntax)
+                    && Model.GetConstantValue(argumentSyntax, CancellationToken) is { HasValue: true, Value: int logLevel }
+                        ? ValidLogLevel.Contains(logLevel)
+                        : true; // Compliant: Some non-constant value is passed as loglevel.
```

Comment:
Alignment is off, should be -4 spaces to get aligned with the 1st condition
```suggestion
                && Language.MethodParameterLookup(invocation, symbol).TryGetNonParamsSyntax(logLevelParameter, out var argumentSyntax)
                && Model.GetConstantValue(argumentSyntax, CancellationToken) is { HasValue: true, Value: int logLevel }
                    ? ValidLogLevel.Contains(logLevel)
                    : true; // Compliant: Some non-constant value is passed as loglevel.
```